pkgs/applications/misc/st/default.nix

```diff
@@ -1,33 +1,32 @@
-{ stdenv, fetchurl, pkgconfig, writeText, libX11, ncurses, libXext, libXft, fontconfig
-, conf ? null, patches ? []}:
+{ stdenv, fetchurl, pkgconfig, writeText, libX11, ncurses, libXext, libXft
+, fontconfig, conf ? null, patches ? []}:
 
 with stdenv.lib;
 
-stdenv.mkDerivation rec {
-  name = "st-0.6";
-  
+let patches' = if isNull patches then [] else patches;
```

Comment:
Patches can be given by user.
We want to also apply our own patch in the code down below (st-fix-deletekey) ([given here](https://github.com/andsild/nixpkgs/blob/29aac270898118a446255ec00540e5ef258f802c/pkgs/applications/misc/st/default.nix#L15))

[Here](https://github.com/andsild/nixpkgs/blob/2e826713a3b94812afd90d4cc8a8cb004dfc186d/pkgs/top-level/all-packages.nix#L14740) however, "patches" is given default null, so we must add the if statement to make it non-null.

The ideal would be to just do `patches = patches ++ [ ./st-fix-deletekey.patch ]` but since we are in a recursive construct, that gives infinite recursion. Therefore the "let" construct outside the rec block